src/main.c

```diff
@@ -31,6 +31,26 @@ do { \
     } \
 } while(0)
    
+void convert_mac(char *input, u_char *output) {
+    int x, y;
+    char data[13] = "AAAAAAAAAAAA0";
+    memset(output, 0, 12);
+    strncpy(data, input, 12);
+
+    for(x=0, y=0; x<=12; x+=2, y++) {
+        if (data[x] >= 'A' && data[x] <= 'F') {
```

Comment:
The logic of this function seems to be flawed:

* The loop checks for x <= 12 while incrementing x by two, so the code will actually write to the 13th byte on the sixth iteration.
* The 0 at the end of `char data[13] = "AAAAAAAAAAAA0";` is not a null byte but the ASCII character 0. The only way (that I know of) of "writing out" the null byte is as a single char `'\0'`. A better way would probably be to simply zero out the entire array with `memset()`.